packages/yew/src/utils.rs

```diff
@@ -1,47 +1,28 @@
 //! This module contains useful utilities to get information about the current document.
 
-use anyhow::{anyhow, Error};
 use std::marker::PhantomData;
-use web_sys::{Document, Window};
+use yew::backend::{Document, NoHostError, NoOriginError, Renderer, RenderingBackend, Window};
 use yew::html::ChildrenRenderer;
 
 /// Returns the current window. This function will panic if there is no available window.
 pub fn window() -> Window {
-    web_sys::window().expect("no window available")
+    Renderer::get_window()
 }
 
-/// Returns the current document.
+/// Returns the current document. This function will panic if there is no available document.
 pub fn document() -> Document {
-    window().document().unwrap()
+    Renderer::get_document()
 }
 
 /// Returns the `host` for the current document. Useful for connecting to the server which serves
 /// the app.
-pub fn host() -> Result<String, Error> {
-    let location = document()
-        .location()
-        .ok_or_else(|| anyhow!("can't get location"))?;
-
-    let host = location.host().map_err(|e| {
-        anyhow!(e
-            .as_string()
-            .unwrap_or_else(|| String::from("error not recoverable")),)
-    })?;
-
-    Ok(host)
+pub fn host() -> Result<String, NoHostError> {
```

Comment:
I guess this is technically a breaking change. Do we want to continue to support anyhow by doing a `#[from]` perhaps?